### DISCUSSION TITLE
Should we continue development on dev branch

### DISCUSSION THREAD

**Discussion by huchenlei** [lllyasviel/stable-diffusion-webui-forge#705] [Polls]
@lllyasviel has been inactive in maintaining forge for about 2 months. Collaborators do not have permission to merge to main now. Should we continue our development in `dev` and let all our users switch to `dev` branch instead?

**Comment by huchenlei** [comment]
I think another option is that we try to merge SD Forge's memory management back to main A1111 repo. I think this might be doable if we can setup good enough memory profiling for torch, and pinpoint the root performance issue in A1111.

**Reply by strawberrymelonpanda** [reply]
> I think another option is that we try to merge SD Forge's memory management back to main A1111 repo

Has anyone talked with @AUTOMATIC1111 (via Discord?) for his opinion on the Forge backend?  Upstreaming these changes would make life easier I feel, but whether any attempt would ultimately be merged really depends on if he's on board.

**Reply by catboxanon** [reply]
@strawberrymelonpanda Yes. Auto's answer was they have no consideration in "using comfy in any way" (ultimately where Forge's memory management and `ldm` module comes from). [Source](https://discord.com/channels/1028645657159938098/1028759674142601366/1141241010769371216) (from the [A1111 dev Discord](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Contributing)).

This discussion that took place on Discord in August 2023 is the reason this fork exists to begin with.

**Reply by strawberrymelonpanda** [reply]
Thanks for the info!  _(I'm not a discord member and wish discussion had never moved there, but oh well.)_

So yeah, sounds like it would come down to improving his / the CompVis(?) implementation.  That'd be wonderful, but I have to admit I'm skeptical after all this time how practical.  Godspeed. 👍

**Comment by slashedstar** [comment]
I mean, everything is reversible anyway; there's no reason not to do it, as is the project is just rotting away for (seemingly) no reason.

**Reply by strawberrymelonpanda** [reply]
I'm on board and voted yes, but sadly I think the biggest reason would be lack of visibility.  Most people aren't going to know to look at "dev2" and as things stand they can't be merged back into Main without @lllyasviel.  Even the ReadMe can't be changed.

I hate to suggest what would be basically a fork of a fork, but...

**Comment by huchenlei** [comment]
I have merged some trivial fix to https://github.com/lllyasviel/stable-diffusion-webui-forge/tree/dev2. Please switch there and try out these patches. Thanks!

**Comment by TheRealPhooney** [comment]
If there had been any indication from @lllyasviel himself that he would ever do anything on it again, I'd vote to wait, but since that isn't the case it feels like a good idea to do something to move forward and I appreciate you stepping up.

Let's face it, a single comment saying he got busy with real life or another project and would or would not be back to continue development could have been done if he wanted to make it assuming he still had internet access, which he did.

It's a shame, but he owes us nothing, this is just the downside of using a product/service/demo at the cutting edge like this.  The upside is that the code is available for others who can grasp it to continue development, taking the wonderful work that @lllyasviel did and running with it.

**Comment by E2GO** [comment]
It's may be some private own projects or something cooking for future for everybody. Check out @lllyasviel activity on private repos. Anyway dev2 branch is a good decision meanwhile and let's have some patience - may be we'll be great surprised by @lllyasviel as usual with his team's projects)
![image](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/1776669/e1339ffa-5ca4-4298-9831-22201fcd877e)

**Comment by altoiddealer** [comment]
This is pretty interesting timing... great minds think alike?

Check out [this Discussion I opened on Auto1111](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/15691) just last night.  No response yet but,...yeah, the inactivity in Forge development is starting to get unnerving, knowing that the next best alternative comes with a significant expense to performance and memory.

The community will be truly blessed if another motivated dev (such as yourself, it seems!) were to save this from extinction.

**Comment by LadyFlames** [comment]
https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/680 if you check this Thread apparently someone got into contact with illya indirectly through discord just scroll to the bottom of that thread

**Comment by gohan2091** [comment]
Can someone show me how to update my Forge to the new Dev2 version?

**Reply by CCpt5** [reply]
If you're familiar w/ how to do a "simple" git pull to update via a cmd window, you'd do the same process but instead of git pull:

git checkout dev2

That should switch you to the dev2 branch. Then you can git pull just to verify you're updated to that. 

"git branch" should show you the list of branches and have a * next to Dev2 meaning you're in sync w/ that one. 

You want to be in the venv setup obv (but you would do that prior to a git pull also so).....

**Reply by gohan2091** [reply]
I have Forge installed via Stability Matrix. Does that matter? so I go to the main Forge folder, in the address bar type CMD then type: venv\scripts\activate.bat ? then I can do git checkout dev2 followed by git pull?

**Reply by gohan2091** [reply]
Still need help. Can someone help me install dev2? I have tried using the instructions CCpt5 wrote and I get an error. Obviously I am doing something wrong :( ![Screenshot 2024-05-07 005429](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/127342824/371cd64b-4a79-4ac8-9f74-33c0163cfb72)

**Reply by CCpt5** [reply]
@gohan2091 Odd. Do a "git pull" (or git fetch) to make sure you're 100% up to date already. I did the process again on my local system (screenshot) and works fine - just shows I"m already on Dev2. Also pulled up the help screen to list commands.

I ran your error through Claude Opus (chatgpt4 alternative) and he seemed to think you're just not up to date so it's not recognizing the request to join "dev2" as "dev2" may not have been create last time you updated.

Hope that's it and it helps.....if not dunno....
![654651515](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/37972078/b3f66b4d-2532-4195-8007-433f1eb567ad)

**Reply by gohan2091** [reply]
![Screenshot 2024-05-07 013946](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/127342824/08fa02a5-c57d-4df9-b056-fcb287b94ca8)

Fixed it. Thanks for your help :)

**Reply by gohan2091** [reply]
@CCpt5 I updated to Dev2 to use Multi-input ControlNet for FaceID. It's added the menu but it's not working. Any ideas?
![Screenshot 2024-05-07 014900](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/127342824/b3ceb36b-b6fb-4cd5-88d5-4f1b43b05916) I tried every Controlnets with Multi-Inputs but nothing is working. EDIT: Actually none of the ControlNet is working at all, not even Single Inputs :(

**Reply by CCpt5** [reply]
As I understand it due to @lllyasviel being away, and the only person who can commit to the main branch, controlnet on Forge is behind and likely doesn't have the most recent functionality. I assume if @huchenlei opts to continue development on another branch (or in some fashion) updates to controlnet would be something that would happen relatively quickly. They do an amazing job implementing/updating controlnet for A1111.

Have to wait for now.

**Reply by gohan2091** [reply]
Why would anyone be using dev2 then if it completely breaks ControlNet? I have no function available in ControlNet, every option results in the same error. Dev2 adds Multi-Input tab but why add that if the entire ControlNet is broken? Posted it in the issue section here: https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/709

**Reply by CCpt5** [reply]
Yea, it's all fluid - I don't know.

To roll back just do the opposite and git checkout main, pull, and should be as you had it.

**Reply by gohan2091** [reply]
I don't mean to sound ungrateful but the main branch lacks Multi-inputs tab and the dev2 adds it... great.... except now the entire ControlNet in dev2 is broken so why add the tab to a broken feature? makes no sense. Yes I have rolled back to the main branch for now so I can use ControlNet again. Thanks for your help.

**Reply by CCpt5** [reply]
Yea,, that's kinda the hope of people in this thread - that updates continue....there's a bunch of stuff that's outdated and bugs that have cropped up....best case is the main developer is lllyasviel ok and comes back to the project. 2nd best case is that huchenlei continues on dev2 or elsewhere (fork?). Until then A1111 is the alternative (or can use in tandem).

**Reply by gohan2091** [reply]
I use to use SDNext as my main but that's become such a mess in my opinion although it is faster. I will continue on with the main branch of Forge for now. Let's hope...

**Comment by thiagojramos** [comment]
If you can continue this, even with a possible return of illv on the horizon, we will be eternally grateful. I tried to make some changes by myself and... well... I had to revert everything back, haha...

**Comment by GavChap** [comment]
Should I do a pr into dev2 for the same commit as I've done for main then?

I've got another one waiting in my fork for the Align Your Steps scheduler as well.

**Reply by wywywywy** [reply]
> Align Your Steps scheduler

That'd be amazing!

**Comment by freecoderwaifu** [comment]
It is curious what lllyasviel might be cooking in his private commits, and at least he seems to be ok if he's active behind the scenes. 

That said, I think the show must still go on in some way to avoid falling behind, and the dev2 branch is a good compromise.

**Comment by devils-shadow** [comment]
I've been refreshing the Forge git page randomly for the past few months, waiting for updates. I never once assumed the project was dead because as it stands, at least for my use-cases, it is rock solid, with top-notch performance and no bugs that have affected my day-to-day usage.

I just assumed that's the reason for a lack of updates, because adding random stuff might lead to performance degradation.
I'd like to believe that llyasviel will return when he has the time. Until then, for anyone wanting to experience a more cutting-edge build, dev2 is the way to go. 

I really hope Forge continues as a project. SD.Next is excellent but simply refuses to allow me to do txt2img upscales due to OOM (diffusers/SDXL), and automatic1111 is slow. Forge has been the best thing to happen to my workflow since I discovered it.

**Comment by MNeMoNiCuZ** [comment]
There are a bunch of quite critical performance updates to the Extra Networks tab that are being worked on for A1111. Once these are fixed, I may switch back to vanilla A1111 just to be able to choose models more effectively when working.

https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15530

I may work in A1111 when setting up prompts, and then switching to Forge for speeds when generating for a long duration.

**Comment by gohan2091** [comment]
> I've been refreshing the Forge git page randomly for the past few months, waiting for updates. I never once assumed the project was dead because as it stands, at least for my use-cases, it is rock solid, with top-notch performance and no bugs that have affected my day-to-day usage.
> 
> I just assumed that's the reason for a lack of updates, because adding random stuff might lead to performance degradation. I'd like to believe that llyasviel will return when he has the time. Until then, for anyone wanting to experience a more cutting-edge build, dev2 is the way to go.
> 
> I really hope Forge continues as a project. SD.Next is excellent but simply refuses to allow me to do txt2img upscales due to OOM (diffusers/SDXL), and automatic1111 is slow. Forge has been the best thing to happen to my workflow since I discovered it.

Is your ControlNet working in dev2? because for me it's completely broken

**Reply by thiagojramos** [reply]
I haven't tested ControlNet after switching to dev2, but after reading about your experience of finding it broken, I went to test it and indeed, it's also giving me errors.

**Reply by gohan2091** [reply]
Thanks for confirming. I just don't understand why the Multi-input tab was added to ControlNet when the entire ControlNet is broken. Kind of pointless. Hope it can be fixed sometime :)

**Reply by thiagojramos** [reply]
I believe the update (to add multi-input) broke the integrated ControlNet in Forge (as far as I remember, the one in main works fine). It's just a guess, but I believe the updated code works for the case of the normal ControlNet (the external extension), but since the integrated one is a bit different, maybe that's the issue.

**Comment by Moussack** [comment]
is @lllyasviel still working on [IC-Light](https://github.com/lllyasviel/IC-Light) ??
hope he's not abandoning Forge.

**Reply by LadyFlames** [reply]
as far as anyone knows he is working on his Private Projects those on his Discord Server would probably know more on what he is doing

**Comment by mweldon** [comment]
Yes, please continue development in dev2. There is no guarantee that the author is ever coming back, despite whatever the rumors might suggest. The alternative is that we get a bunch of forks going in different directions, which is what always happens when there is high demand but no leadership. 

I tried doing a fork and just merging from A1111 but there was too much. I don't know the code well enough to be able to filter through all of that. Maybe a series of smaller merges would be better or maybe a more knowledgeable person could do it.

**Comment by GavChap** [comment]
How do I commit to dev2?

**Reply by huchenlei** [reply]
Open a PR and choose dev2 as base.

**Reply by GavChap** [reply]
I have, but it still saying it needs a review from lllyasviel

**Reply by huchenlei** [reply]
I can review and merge your PR to dev2 without the review from lvmin.

**Reply by GavChap** [reply]
https://github.com/lllyasviel/stable-diffusion-webui-forge/pull/710 if you want to take a look

**Reply by thiagojramos** [reply]
@GavChap If you switch to dev2, could you check for me if controlnet is working in this PR? Thanks :D

**Comment by LadyFlames** [comment]
i will just post this here but i know there is a thread for this does any one know how to Fix the issues in this Thread #727 the issues on that thread are happening to a lot of people Before and after switching to Dev2 those issues partly make Forge unusable at least when used with External&integrated Extensions and all Scripts but if used without any External&Integrated extensions and Scripts it Works Normally and theres still the Problem of Hires Fix being Slower Than usual takes 10hours to do 1 image can anything be done if not then i will use Forge without Extensions etc

**Comment by gohan2091** [comment]
I get that Illyasviel has given his time for free and we should (and are) grateful for Forge but no communication from him? no quick announcement to let us know the current situation? Get our hype up with this UI and then abandon us for an unknown amount of time. Very disappointing that Illyasviel can't spend a couple of minutes writing us a quick message.

**Reply by thiagojramos** [reply]
Just to reiterate, he abandoned the project. I'll use the example of Fooocus, which I closely followed. Last year, when he needed to step away to focus on other projects, he created a post letting everyone know and mentioned he would only merge important PRs from the community. Even now, he left someone in charge of Fooocus and used to respond to people’s posts whenever he could.

But with this project, after that controversial discussion about using something from ComfyUI in Forge (I’m not exactly sure about the details), he just vanished.

To me, it always seemed like this project was more of an experiment for him. Maybe, after that discussion, he decided to drop it for some reason. Unlike Fooocus, which is his "legitimate child," Forge seems to be the "problematic adopted child." Sorry for the metaphor, but that’s what came to mind.

In short, I think he gave up on this project for the reasons I mentioned.

**Comment by itsuxX** [comment]
Hello everyone, can someone please explain how to switch to DEV2? Where do I need to configure something? I'm just a user. Thank you very much.

**Reply by thiagojramos** [reply]
Here: https://gprivate.com/6bepe

<details><summary>Break (click) in case of emergency</summary>
<p>

First, run `git switch dev2` followed by `git pull`. If you encounter the error "Your local changes to the following files would be overwritten by merge", type `git stash`. If the error persists, reset everything with `git reset --hard` and then try `git pull` again. That should solve the issue.

Note: Make a backup of your `webui-user.bat` before using `git reset --hard` because it will reset the file. You can replace the old file with the new one afterward.

![2024-05-27 (2)](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/45890502/09f55ea5-1af4-43b2-a27f-3035a455775d)

</p>
</details>

**Reply by itsuxX** [reply]
Thank you very much

**Comment by ghost** [comment]
I'm not a developer, and I could be completely wrong, so please pardon me if I am. Doesn't it make more sense to integrate Forge memory management in A1111 instead? To my understanding, multidiffusion and Tiled Vae already work in a similar manner.

**Reply by thiagojramos** [reply]
https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/166

**Comment by amitsandhel** [comment]
So i'm a bit confused as I too want to help develop this or would like to spend time but not sure if it would be helpful to the community ?
so is forge kind of dying and the main developer ended up building fooocus which is the one he is investing his time on ?
Also seems we can't take the speed advantage forge has and put into into A1111 as those developers don't want to do that for some odd reason ?
I mean is the backend of forge so different that nothing can be even ported to a1111 ?

**Reply by CCpt5** [reply]
> So i'm a bit confused as I too want to help develop this or would like to spend time but not sure if it would be helpful to the community ? so is forge kind of dying and the main developer ended up building fooocus which is the one he is investing his time on ? Also seems we can't take the speed advantage forge has and put into into A1111 as those developers don't want to do that for some odd reason ? I mean is the backend of forge so different that nothing can be even ported to a1111 ?

Can't speak to all of this, but Fooocus came well before Forge. 

Also, [huchenlei](https://github.com/huchenlei) (OP) recently reviewed the code for A1111 and found new optimizations that increased performance speeds to almost (or better than) Forge. They're in the PR section over there and you can add them manually if you so desire: https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15821

I think we're all hoping development on Forge restarts or is at least "allowed to continue" either under a new branch or a new fork. So far even the idea mentioned here hasn't gotten any traction. Perhaps the imminent release of SD3 will change this.

**Reply by MNeMoNiCuZ** [reply]
Thanks for sharing.
That 15821 sounds awesome!

I'm such a fan of Forge, but when that comes in and if it's good, and this one (https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15530), I think I'll need to head back to A1111. It's so frustrating to be unable to use the Extra Networks view. It got completely unusable if you have 10k+ models when the tree list got introduced. Hopefully this makes it speedy speedy!

**Reply by amitsandhel** [reply]
hmm I'll test out that Pr but if that PR does get added would forge have any use case left then ?\
I want to actually develop and code some stuff for this and learn how to do it but I would like to do it in a project which won't be "dead" so to speak or won't go anywhere

**Reply by minsky91** [reply]
#15821 sounds like a great news! So there is a hope to lift Automatic to Forge's level of speed,  after all. What about memory management, another Automatic's week spot, is it being addressed in #15821? Also, I am afraid we are forgetting here the main issue with Automatic, next to its poor speed: it's a messy piece of software. Every now and then I run into a crash, or a cryptic error message, or just a paralysis of Automatic, when using it with high resolution images. If someone really is determined to straighten out Automatic, this should become a priority as well. If Automatic continues being such a mess, I will look elsewhere, where bugs are still being fixed (if not Forge, then SD.Next, StableSwarm, and even Comfy).

**Reply by amitsandhel** [reply]
> #15821 sounds like a great news! So there is a hope to lift Automatic to Forge's level of speed, after all. What about memory management, another Automatic's week spot, is it being addressed in #15821? Also, I am afraid we are forgetting here the main issue with Automatic, next to its poor speed: it's a messy piece of software. Every now and then I run into a crash, or a cryptic error message, or just a paralysis of Automatic, when using it with high resolution images. If someone really is determined to straighten out Automatic, this should become a priority as well. If Automatic continues being such a mess, I will look elsewhere, where bugs are still being fixed (if not Forge, then SD.Next, StableSwarm, and even Comfy).

Here is some of the testing I did. I have a Nvidia GTX1050Ti Graphic card with only **4Gb** of VRAM. i don't have a lot of VRAM and I am on Linux using PopOs a flavor of Ubuntu. I was trying to run SDXL checkpoints and generate images with that.

- **SdNExt**: I tried sdnext there using a different backend called diffusers. They are fast Currently there getting a lot of errors since they broke off from A1111 and now are a new thing. For me sdnext doesn't work. With respect to SDXL it crashes it can generate a basic image but I can't use face restore and even highres fix since I go out of memory. The same is true with using SD1.5 with a 512x768 imge I couldn't use their highres fix or diffusion fix and the images were different from what civitai or A1111 or forge generates. I ended up uninstalling and deleteing SDNext sadly. 

- **Comfy:** I did try comfy last night. It was easy to use actually and ironically I did see that the main time spent by the process was in the sampler where it was using the sampler to get an image out the door. For me using euler A for 24 steps on my nvidia gtx1050Ti card with 4Gb of Vram it was doing about 36 sec/iteration.  Do note this was for a SDXL checkpoint an aaaautism checkpoint and I was generating a simple anime girl picture and pose. I did run comfy with the --lowvram command but saw no real change or impact in performance. i am not using controlnet or anything for this test. Just 3 loras and a basic sampler and checkpoint. i ended up uninstalling this too as it was too slow. 

- **Forge**: I have tried forge and use forge to this day. Ironically with there OOm and freeU and even with their optimization I can run SD1.5 checkpoints and models at a a speed of 1-2sec/iterations. With respect to controlnet I can even use controlnet on a SD1.5 checkpoint and model easily with little speed loss it becomes about 8 seconds/iteration. 
      - Furthermore I can even run SDXL/pony checkpoints at about 24sec/iteration with forge. I typically use Euler A with a step count of 24. I can even use Adetailer as well to fix the faces. 
  
One new thing I discovered are the **sdxl lightning** checkpoints which by using the LCM sampler you can generate an image in 9 steps which is an interesting approach it takes 24 second but only gotta suffer for 4-9 rounds. 

I have not tried a111 yet but the first time i ever installed it was months ago around january of 2024 and it was slow like 40sec/iteration but I can't say anything conclusive about that. 

So far it seems forge is the best for low graphic card users or at least the best in terms of performance. So it could be worth the time to invest in dev2 and build new features or possibly think of new ways to get more bang for our buck.

Hope this helps.

**Reply by Moussack** [reply]
> > #15821 sounds like a great news! So there is a hope to lift Automatic to Forge's level of speed, after all. What about memory management, another Automatic's week spot, is it being addressed in #15821? Also, I am afraid we are forgetting here the main issue with Automatic, next to its poor speed: it's a messy piece of software. Every now and then I run into a crash, or a cryptic error message, or just a paralysis of Automatic, when using it with high resolution images. If someone really is determined to straighten out Automatic, this should become a priority as well. If Automatic continues being such a mess, I will look elsewhere, where bugs are still being fixed (if not Forge, then SD.Next, StableSwarm, and even Comfy).
> 
> Here is some of the testing I did. I have a Nvidia GTX1050Ti Graphic card with only **4Gb** of VRAM. i don't have a lot of VRAM and I am on Linux using PopOs a flavor of Ubuntu. I was trying to run SDXL checkpoints and generate images with that.
> 
> * **SdNExt**: I tried sdnext there using a different backend called diffusers. They are fast Currently there getting a lot of errors since they broke off from A1111 and now are a new thing. For me sdnext doesn't work. With respect to SDXL it crashes it can generate a basic image but I can't use face restore and even highres fix since I go out of memory. The same is true with using SD1.5 with a 512x768 imge I couldn't use their highres fix or diffusion fix and the images were different from what civitai or A1111 or forge generates. I ended up uninstalling and deleteing SDNext sadly.
> * **Comfy:** I did try comfy last night. It was easy to use actually and ironically I did see that the main time spent by the process was in the sampler where it was using the sampler to get an image out the door. For me using euler A for 24 steps on my nvidia gtx1050Ti card with 4Gb of Vram it was doing about 36 sec/iteration.  Do note this was for a SDXL checkpoint an aaaautism checkpoint and I was generating a simple anime girl picture and pose. I did run comfy with the --lowvram command but saw no real change or impact in performance. i am not using controlnet or anything for this test. Just 3 loras and a basic sampler and checkpoint. i ended up uninstalling this too as it was too slow.
> * **Forge**: I have tried forge and use forge to this day. Ironically with there OOm and freeU and even with their optimization I can run SD1.5 checkpoints and models at a a speed of 1-2sec/iterations. With respect to controlnet I can even use controlnet on a SD1.5 checkpoint and model easily with little speed loss it becomes about 8 seconds/iteration.
>   - Furthermore I can even run SDXL/pony checkpoints at about 24sec/iteration with forge. I typically use Euler A with a step count of 24. I can even use Adetailer as well to fix the faces.
> 
> One new thing I discovered are the **sdxl lightning** checkpoints which by using the LCM sampler you can generate an image in 9 steps which is an interesting approach it takes 24 second but only gotta suffer for 4-9 rounds.
> 
> I have not tried a111 yet but the first time i ever installed it was months ago around january of 2024 and it was slow like 40sec/iteration but I can't say anything conclusive about that.
> 
> So far it seems forge is the best for low graphic card users or at least the best in terms of performance. So it could be worth the time to invest in dev2 and build new features or possibly think of new ways to get more bang for our buck.
> 
> Hope this helps.

Yeah forge is excellent for low end hardware, I've tested it with my gtx 970 4GB i can generate 3215x2200 img below with hiresfix with reasonable amount of time (7 minutes). It's impossible to do that in A1111

![00006-277103266](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/69501089/758be07e-4d4e-4732-85d1-198ec934ec39)

**Reply by BNP1111** [reply]
can i ask how to do so great img,thank you

**Comment by aolko** [comment]
Can you please update controlnet? Interested due to composition ip-adapter

**Reply by amitsandhel** [reply]
> Can you please update controlnet? Interested due to composition ip-adapter

That's the question how no one here has access to the permissions to push to master. We would need some way to organize ourselves to work on this assuming there is worth in doing so and this is doing something both a1111 and comfyui can't do?

**Reply by aolko** [reply]
> > Can you please update controlnet? Interested due to composition ip-adapter
> 
> That's the question how no one here has access to the permissions to push to master. We would need some way to organize ourselves to work on this assuming there is worth in doing so and this is doing something both a1111 and comfyui can't do?

**dev2 branch, come on**

**Comment by amitsandhel** [comment]
So is A1111 still slow performance wise after all this time and is anyone using comfy and knows what speeds their getting from say a weak graphic card 
maybe we can do some speed comparisons to see how forge is still faring ?

**Reply by ghost** [reply]
IMHO, stay away from comfy. It's buggy, poorly written software by someone who has no idea what they're doing (node-based UI without built-in custom node management). It's slowly being abandoned by the community. You'll spend a large amount of time just setting up your render, whereas it only takes one or two clicks in A1111 with much better results.

**Reply by aolko** [reply]
however **a lot** of people port shit to nodes from papers, no such love for a1111/forge :(

**Reply by ghost** [reply]
I think that's because when custom node authors abandon their projects (and 90% of them already have, especially the most promising ones), this "paper" feature stops working after updates and causes many problems for other nodes, making this feature absolutely unusable.

**Reply by amitsandhel** [reply]
I tried A1111 today sadly it just didn't work on my 4gb of vram it crashed my laptop instead forcing me to do a force hard shutdown and restart lol 

Forge is still better at least both sdxl and 1.5 work now if only we can figure out which feature and improvements we wish to add and have maybe we can get coding

**Reply by aolko** [reply]
> I tried A1111 today sadly it just didn't work on my 4gb of vram it crashed my laptop instead forcing me to do a force hard shutdown and restart lol 
> 
> Forge is still better at least both sdxl and 1.5 work now if only we can figure out which feature and improvements we wish to add and have maybe we can get coding 

How about stopping ui lag with extensions installed? https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/716#discussioncomment-9639984

**Comment by johiny** [comment]
SD forge is amazing lllyasviel did an incredible work on memory management, I don't know what kind of memory management  A1111 has but it literally crashed my system on Linux with ROCM. 
I have tested SD Next, Comfy UI, A1111 and all has some kind of weird memory behaviors, like crash the whole system or eating up so many resources I guess majority vram? that secondary screens become black.
But Forge has given me the most stable experience, and for me after changing to AMD and passing through the literal hell to make SD work.
I really appreciate that stable behavior, so I think work on dev branch meanwhile lllyasviel comes back or not is a nice idea.

**Comment by CantainPike** [comment]
You may try StableSwarm still in beta which is works faster than on my RTX 3060 12GB VRAM, Forge is one image generation for SDXL 17.3 seceond, StableSwarm 15.6 Second Slightly faster. I wonder how does it works on low VRAM machines. Keep in mind  StableSwarm still in beta some features are under developement.

**Reply by amitsandhel** [reply]
hmm but stableswarm is built In C# as I looked at the codebase uses dockers too so I don't know if that would be useful since dockers from my beginner's experience eats up lots of disk space and you need dotnet8 as well

**Reply by aolko** [reply]
it's ugly AF and has a bad UX

**Reply by altoiddealer** [reply]
> it's ugly AF and has a bad UX

Been lurking this thread, but just want to pop in an mention that the author of StableSwarm (or at least the key dev / face of the project) is esteemed community member McMonkey.  Besides, that comes across a bit disrespectful for anything open source, freely gifted to us

**Reply by aolko** [reply]
> > it's ugly AF and has a bad UX
> 
> Been lurking this thread, but just want to pop in an mention that the author of StableSwarm (or at least the key dev / face of the project) is esteemed community member McMonkey. Besides, that comes across a bit disrespectful for anything open source, freely gifted to us

what truly is disrespectful is denying user contributions

**Reply by aolko** [reply]
oh and remind me who abandoned stablestudio ui real quick

**Comment by amitsandhel** [comment]
It seems after some research we have conducted we (the small group that interacted in this discussion post) could conclude forge has value and we should code and work on it.
 
It still has the fastest speed and works beautifully on low graphic cards and even high end cards can generate higher quality images that they can't do otherwise using other tools. 

So then we have two questions 
1) do we have any tutorials to read on how to code or write some code for this 
2) who wishes to manage this branch or do we just fork it and make our own branches and make pr's for it. I doubt anyone should be self inserting their own code into dev2 ?
3) do we create some more appropiate communication stream like sdnext did using discord or something ?
4) anyone got in contact with the main author and asked him about forge ?

**Reply by aolko** [reply]
let's fork this and make a discord server for it

**Reply by amitsandhel** [reply]
Sounds like a plan let's get cracking 
Autobots roll out ! 
What's the discord server btw ?

**Comment by PanduZen** [comment]
@huchenlei 

Now that the announcement has happened, are you planning to keep updating the dev2 branch?, I don't mean to implement new complex things, but to fix bugs and update basic things for those who want to keep using forge?

**Reply by obs945** [reply]
> @huchenlei
> 
> Now that the announcement has happened, are you planning to keep updating the dev2 branch?, I don't mean to implement new complex things, but to fix bugs and update basic things for those who want to keep using forge?

Can someone make a simple tutorial to help us switch from main to dev2 branch ?
I tried "git checkout dev2 && git pull" but in vain. The process was terminated during modules/launch_utils.py modification.

**Reply by PanduZen** [reply]
> > @huchenlei
> > Now that the announcement has happened, are you planning to keep updating the dev2 branch?, I don't mean to implement new complex things, but to fix bugs and update basic things for those who want to keep using forge?
> 
> Can someone make a simple tutorial to help us switch from main to dev2 branch ? I tried "git checkout dev2 && git pull" but in vain. The process was terminated during modules/launch_utils.py modification.

use stability matrix

**Reply by hoodady** [reply]
Now here's how to switch, nice and easy:

Go to the root directory of your Forge installation (i.e. whichever folder has "webui-user.bat" in it)

Open a terminal window inside this directory

git pull (updates Forge if it isn't already)

git fetch origin (fetches all branches)

git switch -c dev2 origin/dev2 (switches to the dev2 branch)

Done!

**Reply by hoodady** [reply]
If you'd ever like to switch back, just run git switch main from the terminal inside the same directory

**Comment by MNeMoNiCuZ** [comment]
Let's go! Forge2 <3

**Comment by aolko** [comment]
Wait, that's not "resolved", is it?

**Reply by altoiddealer** [reply]
Perhaps Ill has now returned to save us all.

(I won’t hold my breath)

**Reply by aolko** [reply]
> Perhaps Ill has now returned to save us all.
> 
> (I won’t hold my breath)

nope

**Comment by prog-ape** [comment]
@huchenlei Hello, will there be further updates? I'm eagerly looking forward to them. Even though I have seen your patches in the web UI, the execution speed during API calls in the web UI is still very slow compared to Forge.

**Reply by aolko** [reply]
here's one:
move to https://github.com/Panchovix/stable-diffusion-webui-reForge
